erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -1305,7 +1305,7 @@ def item_query(doctype, txt, searchfield, start, page_len, filters):
 		if not field in searchfields
 	]
 
-	query_filters = {"disabled": 0, "ifnull(end_of_life, '5050-50-50')": (">", today())}
+	query_filters = {"disabled": 0, "end_of_life": (">", today())}
```

Comment:
This isn't correct. if end of life is not specified than it's considered to have infinite end of life.